spec/features/user_can_register_with_spotify_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.feature 'A user registers an account through Spotify' do
+  describe 'As a visitor, when I go to the homepage' do
+    describe 'and I scroll to the register/login section' do
+      it 'I should see a register button' do
+        visit '/'
+        expect(page).to have_button("Register with Spotify")
+      end
+      describe 'when I click on the register button' do
+
+        describe 'and I sign in successfully on Spotify' do
+          before :each do
+            stub_oauth_registration
+            Rails.application.env_config['omniauth.auth'] = OmniAuth.config.mock_auth[:spotify]
+            visit '/'
+            click_button "Register with Spotify"
+          end
+
+          it 'should redirect me to the dashboard' do
+            expect(current_path).to eq('/')
+          end
+
+          it 'the navigation bar should now include Logout' do
+
```

Comment:
I really like how many `it` blocks you have. It makes it very easy to walk through the test and understand what's going on. I do think the spaces after the `it` line are unnecessary stylistically (lines 25, 41, 39).